### DISCUSSION TITLE
Xcode 15

### DISCUSSION THREAD

**Discussion by marchbold** [airsdk/Adobe-Runtime-Support#2839] [General]
As Apple has released Xcode 15 we are getting ready to update our extensions for iOS 17 support etc

Not surprisingly there are some build issues, and I'll start reporting what I find as I go, but here's the first one:

- https://github.com/airsdk/Adobe-Runtime-Support/issues/2838

What's the schedule for updating to Xcode 15 / iOS 17  look like from the AIR SDK perspective? Would be good to have a rough time frame so we can plan ANE updates :)

**Comment by ajwfrost** [comment]
Hi

We had started looking at this, there are a few different threads to it:

1) making sure that the existing tools work if you update Xcode .. it used to be independent but of course now we're using Apple's tools for some things when you're running on macOS, hence the issue with their linker changing to remove a command-line option. This is in progress and should be released shortly, we have two other issues I'd like to get into the same build but certainly we'll get this out by the end of the month (there's a workaround for it..)

2) building the AIR runtimes using the latest Xcode etc - generally this has always taken longer to do because of the changes Apple keep making to things. I'm hoping this time that the efforts we put in for Xcode 14 will actually mean less of an increment to get Xcode 15 support, but this is what's currently under investigation.. This part also is where we build all the 'stub' files so this is going to be needed before people are able to take advantage of any new APIs in the iPhoneOS SDK 17.

3) updating the build tools so that the outputs appear as if they're built with the latest Xcode/SDK etc. This is what's needed for app store submissions and is something we can set up now that Xcode 15 and the v17 SDKs are out of beta..

So short term, by the end of this month we will have a release with (1) and (3), and I'm hoping (2) won't be too far behind.

thanks

**Reply by marchbold** [reply]
Brilliant! Thanks for the update, definitely welcomed news!

**Reply by raresn** [reply]
@marchbold I am guessing the contacts ANE will have to wait for this release of AIRSDK as well, right?

**Reply by marchbold** [reply]
@raresn I don't believe so, unless there are new features in iOS 17 you were hoping for?

**Reply by raresn** [reply]
@marchbold i don't know of any new features in iOS 17 - at this point we are just trying to get a fix for this as soon as possible to counter all the support tickets we got on this :) If you say that it's possible to get fied without waiting for the new AIRSDK that would be super helpful!

**Reply by ajwfrost** [reply]
@raresn what fix are you looking for? not 100% sure from the above context, I feel I'm missing a reference to a bug report! Unless it's just that you'd updated to Xcode 15 and are then hitting the issue mentioned in #2838 ? (there's a workaround for that one)

**Reply by raresn** [reply]
@ajwfrost sry for hijacking this thread and creating confusion
My comment was related to an ANE from distriqt. (Contacts ane, which doens't fully work on ios 17)
